mempool/reactor.go

```diff
@@ -31,13 +31,25 @@ const (
 	maxActiveIDs = math.MaxUint16
 )
 
+// MempoolWithWait extends the standard Mempool interface to allow reactor to
+// wait for transactions and iterate on them once there are any. Also it
+// includes ReapMaxTxs function, which is useful for testing.
+//
+// UNSTABLE
+type MempoolWithWait interface {
+	Mempool
+	TxsFront() *clist.CElement
+	TxsWaitChan() <-chan struct{}
+	ReapMaxTxs(n int) types.Txs
```

Comment:
I find it a bit odd to introduce this in a public interface if this *only* for testing. This is not test-driven development IMHO (more like test-only API design).

OK, `ReapMaxTxs` is used in testing *and* in `UnconfirmedTxs` (the latter is the only actual use-case). This method should maybe be part of the standard Mempool then?